src/lib/plot/axis/x-axis.js

```diff
@@ -19,13 +19,15 @@
 // THE SOFTWARE.
 
 import React from 'react';
+
 import Axis from './axis';
-import {ORIENTATION} from '../../utils/axis-utils';
+import {DIRECTION, ORIENTATION} from '../../utils/axis-utils';
 
 const {TOP, BOTTOM} = ORIENTATION;
 
+const {axisType, ...axisPropsWithoutType} = Axis.propTypes;
 const propTypes = {
-  ...Axis.propTypes,
```

Comment:
You can do custom validators, an even better option here might be to return an Error if axisType is specified when using x-axis:

https://facebook.github.io/react/docs/typechecking-with-proptypes.html